x/ibc/03-connection/types/querier.go

```diff
@@ -15,10 +15,16 @@ const (
 	QueryClientConnections = "client_connections"
 )
 
+// IdentifiedConnectionEnd defines the union of a connection end & an identifier.
+type IdentifiedConnectionEnd struct {
+	ConnectionEnd
```

Comment:
I think (?) this should use a pointer `*ConnectionEnd`. I'd also double-check that the JSON field is "connection_end" instead of `"ConnectionEnd"`

Reply:
Ah, the idea was to embed the JSON fields - does that not work?

Reply:
This is working on the relayer fwiw, but would be helpful to add a json tag: `json:"connection_end"`

Reply:
```
[
  {
    "ConnectionEnd": {
      "state": "OPEN",
      "client_id": "ibconeclient",
      "counterparty": {
        "client_id": "ibczeroclient",
        "connection_id": "ibczeroconnection",
        "prefix": {
          "type": "ibc/commitment/MerklePrefix",
          "value": {
            "key_prefix": "aWJj"
          }
        }
      },
      "versions": [
        "1.0.0"
      ]
    },
    "identifier": "ibconeconnection"
  }
]

```

Reply:
Ah my bad - updated!

Reply:
wouldn't it be more clean to add the identifier to the `IdentifiedConnectionEnd` flattened?
```json
[
  {
      "identifier": "ibconeconnection"
      "state": "OPEN",
      "client_id": "ibconeclient",
      "counterparty": {
        "client_id": "ibczeroclient",
        "connection_id": "ibczeroconnection",
        "prefix": {
          "type": "ibc/commitment/MerklePrefix",
          "value": {
            "key_prefix": "aWJj"
          }
        }
      },
      "versions": [
        "1.0.0"
      ]
  }
]
```

Reply:
That was what I originally tried to do - how does one do that in Go?